src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigratePackageDependenciesAndToolsRule.cs

```diff
@@ -67,12 +75,57 @@ public void Apply(MigrationSettings migrationSettings, MigrationRuleInputs migra
                     migrationRuleInputs.OutputMSBuildProject,
                     targetFramework.FrameworkName, 
                     targetFramework.Dependencies,
-                    migrationRuleInputs.ProjectXproj); 
+                    migrationRuleInputs.ProjectXproj);
+
+                if (defaultProjectTargetFramework.IsDesktop())
+                {
+                    allDependencies.AddRange(targetFramework.Dependencies);
+                }
+            }
+
+            // Automatically add the same desktop references that the build for project.json does.
+            // This ensures if the project.json build succeeded then the migrated msbuild one 
+            // will succeed too. See src\Microsoft.DotNet.ProjectModel\Resolution for more details.
+            if (defaultProjectTargetFramework.IsDesktop())
+            {
+                List<ProjectLibraryDependency> autoAddDesktopDependencies = new List<ProjectLibraryDependency>();
+
+                // MSBuild targets automatically add a reference to mscorlib
+                // AddIfMissing(autoAddDesktopDependencies, allDependencies, "mscorlib");
+                AddIfMissing(autoAddDesktopDependencies, allDependencies, "System");
```

Comment:
This isn't the right logic here. The existing dependencies can have conditions that are different from the reference you want to inject (conditions are based on tfm). And the injected references should have a condition added so they're only used in the tfm they're being injected for (every desktop tfm in the project).